### ISSUE TITLE
🐛 TinaCMS - building with caching enabled can break pages

### ISSUE THREAD

**Issue by wicksipedia** [tinacms/tinacms#4791]
Raised by @PothieuG

Repro repo privately available - https://github.com/SSWConsulting/NDC_Website
 
I've got issue with Tina :
I updated NDC version to the latest version of tinacms and tina cli (2.2.3 and (1.6).
I can have access to the website, but some pages are not working (the static ones I guess), and in the logs on Vercel I've got this: 
 
```
Error: Cannot find module 'tinacms/dist/cache'
Require stack:
- /var/task/.next/server/chunks/738.js
- /var/task/.next/server/webpack-runtime.js
- /var/task/.next/server/app/[location]/[filename]/page.js
- /var/task/node_modules/next/dist/compiled/next-server/server.runtime.prod.js
- /var/task/___next_launcher.cjs
- /opt/rust/nodejs.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1145:15)
    at /var/task/node_modules/next/dist/compiled/next-server/server.runtime.prod.js:11:28979
    at Module._load (node:internal/modules/cjs/loader:986:27)
    at /opt/rust/nodejs.js:1:11612
    at Function.cr (/opt/rust/nodejs.js:1:11986)
    at e.<computed>.re._load (/opt/rust/nodejs.js:1:11582)
    at Module.require (node:internal/modules/cjs/loader:1233:19)
    at u.require (/var/task/node_modules/next/dist/compiled/next-server/server.runtime.prod.js:11:29195)
    at require (node:internal/modules/helpers:179:18)
    at new l (/var/task/.next/server/chunks/738.js:35:19804) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/var/task/.next/server/chunks/738.js',
    '/var/task/.next/server/webpack-runtime.js',
    '/var/task/.next/server/app/[location]/[filename]/page.js',
    '/var/task/node_modules/next/dist/compiled/next-server/server.runtime.prod.js',
    '/var/task/___next_launcher.cjs',
    '/opt/rust/nodejs.js'
  ],
  page: '/melbourne/tickets'
}
```
 
But it's working fine on local environment.
 
I tried a different version (2..2.1 and 1.6.0), but the same error occured.
 
I rolled back to the version it used to work : 1.6.4 and 1.5.46 (and rolled back modifications in the code), but now, the deployment crash at build time:
 
```
Failed to compile.
./tina/__generated__/client.ts:3:38
Type error: Object literal may only specify known properties, and 'cacheDir' does not exist in type 'TinaClientArgs<{ contentQuery(variables: Exact<{ relativePath: string; }>, options?: { branch?: string | undefined; fetchOptions?: Omit<RequestInit | undefined, "body" | "method"> | undefined; } | undefined): Promise<...>; ... 20 more ...; categoryConnection(variables?: Exact<...> | undefined, options?: { ...; } | u...'.
  1 | import { createClient } from "tinacms/dist/client";
  2 | import { queries } from "./types";
> 3 | export const client = createClient({ cacheDir: '/vercel/path0/tina/__generated__/.cache/1725953292440', url: 'https://content.tinajs.io/1.5/content/a0345e5c-1c0e-4940-8aa9-5cefb4b80058/github/develop', token: '1af65195ac1be28efef1c3af9c45c5f0d1677afd', queries, errorPolicy: 'include' });
    |                                      ^
  4 | export default client;
  5 |   
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Command "yarn run build" exited with 1
```
 a
 
It's still working on local environment.

**Comment by kldavis4** [comment]
this should fix it https://github.com/tinacms/tinacms/pull/4803

**Comment by kldavis4** [comment]
PR updated to also fix #4788 cc @JackDevAU

**Comment by JackDevAU** [comment]
✅ PR Has been merged - https://github.com/tinacms/tinacms/pull/4803 

Fix is available in `tinacms` version **2.2.6** & `@tinacms/cli` version **1.6.6**
